lib/auth/auth.go

```diff
@@ -1671,6 +1671,21 @@ func (a *Server) SetAccessRequestState(ctx context.Context, params services.Acce
 	return trace.Wrap(err)
 }
 
+func (a *Server) GetAccessCapabilities(ctx context.Context, req services.AccessCapabilitiesRequest) (services.AccessCapabilities, error) {
+	var caps services.AccessCapabilities
+	if req.RequestableRoles {
+		v, err := services.NewRequestValidator(a, req.User)
+		if err != nil {
+			return services.AccessCapabilities{}, trace.Wrap(err)
+		}
+		caps.RequestableRoles, err = v.GetRequestableRoles()
```

Comment:
What do you think about encapsulating this logic into the validator. I don't think the handler here should be deciding what parts of `services.AccessCapabilities` should be filled out.

Reply:
While that would make sense with the current iteration, it probably wouldn't work in the long term.  With some of the upcoming changes we have planned, new fields (e.g. `ApprovableRoles`) are going to need to be supported.  The request validator is meant to specifically serve as a helper for evaluating permissions around request creation, so I'd prefer to keep it separate from the more general concept of querying the auth server about capabilities.